Tests/UnitTests/Purchasing/Purchases/PurchasesConfiguringTests.swift

```diff
@@ -92,6 +92,68 @@ class PurchasesConfiguringTests: BasePurchasesTests {
         expect(Purchases.shared.finishTransactions) == true
     }
 
+    func testUserIdIsSetWhenConfiguringWithUserID() {
+        let purchases = Purchases.configure(
+            with: .init(withAPIKey: "")
+                .with(appUserID: Self.appUserID)
+        )
+        expect(purchases.appUserID) == Self.appUserID
+    }
+
+    func testUserIdIsSetToAnonymousWhenConfiguringWithEmptyUserID() {
+        self.deviceCache.userIDStoredInCache = nil
+
+        let purchases = Purchases.configure(
+            with: .init(withAPIKey: "")
+                // This test requires no previously stored user
+                .with(userDefaults: .init(suiteName: UUID().uuidString)!)
+                .with(appUserID: "")
+        )
+        expect(purchases.appUserID).toNot(beEmpty())
+        expect(IdentityManager.userIsAnonymous(purchases.appUserID))
+            .to(beTrue(), description: "User '\(purchases.appUserID)' should be anonymous")
+    }
+
+    func testUserIdOverridesPreviouslyConfiguredUser() {
+        // This test requires no previously stored user
+        let userDefaults: UserDefaults = .init(suiteName: UUID().uuidString)!
+
+        let newUserID = Self.appUserID + "_new"
+
+        _ = Purchases.configure(
+            with: .init(withAPIKey: "")
+                .with(userDefaults: userDefaults)
+                .with(appUserID: Self.appUserID)
+        )
+        Purchases.clearSingleton()
+        let purchases = Purchases.configure(
+            with: .init(withAPIKey: "")
+                .with(userDefaults: userDefaults)
+                .with(appUserID: newUserID)
+        )
+
+        expect(purchases.appUserID) == newUserID
+    }
+
+    func testNilUserIdIsIgnoredIfPreviousUserExists() {
```

Comment:
@vegaro @aboedo this is what I was talking about. To me this seems like a bug, but I wanted to at least document the current behavior.

Reply:
This is the same as in Android so I think it was designed like this, but I am not sure. @aboedo might know better since he worked in Identity v2. I think it's to make cases like the following work better:

- configure with nil -> anonymous is generated
- login with for example a Firebase ID
- configure with nil -> it will reuse Firebase ID

Reply:
Cool 👍🏻 
Well at least we have a test for this now.